### ISSUE TITLE
'async \n export' falsely reports error

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#872]
The following is legal code (according to Node.js), but quick-lint-js rejects it:
```javascript
let async;
async
export function f() {}
```

**Comment by AlekLefebvre** [comment]
To clarify, these would be correct ?

`let async;
async // variable use. ASI
export function f() {}`

`async // async is undefined. ASI
export function f() {}`

`async export function f() {}  // Should be export async`

If so we can fix this by adding this 
```
if (this->peek().has_leading_newline) {
    v.visit_variable_use(async_token.identifier_name());
    break;
}
```
before this [line](https://github.com/quick-lint/quick-lint-js/blob/7fb7cac7ebd352a0558147170ad835245bf833f1/src/quick-lint-js/fe/parse-statement.cpp#L279) in PR #870 

Should I add it to the same PR or wait for it to be merged?

**Comment by strager** [comment]
> To clarify, these would be correct ?

Yes.

> Should I add it to the same PR or wait for it to be merged?

It doesn't matter to me.
